spec/features/announcement_management_spec.rb

```diff
@@ -0,0 +1,143 @@
+require 'rails_helper'
+
+describe 'Announcement management' do
+  subject { page }
+
+  let!(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+    let!(:user) { create(:user, role: :administrator) }
+    let!(:course) { create(:course) }
+
+    before do
+      login_as(user, scope: :user)
+    end
+
+    describe 'announcement creation' do
+      before { visit new_course_announcement_path(course) }
+
+      context 'with invalid information' do
+        before { click_button 'Create' }
+
+        it 'stays on the same page' do
+          expect(page).to have_button('Create')
+        end
+
+        it 'shows errors' do
+          expect(page).to have_css('div.has-error')
+        end
+      end
+
+      context 'with valid information' do
+        let(:announcement) { build(:course_announcement, course: course) }
+
+        before do
+          fill_in 'course_announcement_title',    with: announcement.title
+          fill_in 'course_announcement_content',  with: announcement.content
+        end
+
+        it 'creates an announcement' do
+          expect { click_button 'Create' }.to change(Course::Announcement, :count).by(1)
+        end
+
+        context 'after creation' do
+          before { click_button 'Create' }
+
+          it 'shows the success message' do
+            expect(page).to have_selector('div',
+                                          text: I18n.translate('course.announcements.create.notice',
+                                                               title: announcement.title))
+          end
+
+          it 'redirects the user to the index page' do
+            expect(current_path).to eq(course_announcements_path(course))
+          end
+        end
+      end
+    end
+
+    describe 'announcement editing' do
+      let!(:announcement) { create(:course_announcement, course: course) }
+
+      before { visit edit_course_announcement_path(course, announcement) }
+
+      context 'page rendering' do
+        it { is_expected.to have_field('course_announcement_title', with: announcement.title) }
+        it { is_expected.to have_field('course_announcement_content', with: announcement.content) }
+        it { is_expected.to have_field('course_announcement[valid_from]',
+                                       with: announcement.valid_from) }
+        it { is_expected.to have_field('course_announcement[valid_to]',
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.